guides/Circle CI 2.md

```diff
@@ -0,0 +1,195 @@
+# Circle CI 2.0 configurations for Rails project
+
+## Basic Concepts
+
+CircleCI 2.0 operates on a differently structurized YAML file located under `.circleci/config.yml`.
+
+It's safe to say that every config should contain at least three keywords: `version, jobs`.
+Initially, it should start looking something like this:
+```
+---
+version: 2
+jobs:
+    build:
+        working_directory: ~/your-app-name
+    steps:
+        - checkout
+```
+This intital setup is not going to work. But it’s the basic shell of the config file we’ll need.
+
+## Basic Configurations.
+
+### Select image for Docker executor
+Which will compose the environment from a set of Docker images [CircleCI provide set of pre-built images that we’ll use. ](https://hub.docker.com/r/circleci/ruby/tags/)
+These cover lots of different Ruby versions, with variations for common requirements like a browser and/or JavaScript.
+```
+jobs:
+    build:
+        working_directory: ~/your-app-name
+        docker:
+            - image: circleci/ruby:2.5.1-node-browsers
+            environment:
+                RAILS_ENV: test
+```
+This will set Docker executor to run 2.5.1 ruby version with browser in the box.
+
+### Set up database configurations
+Same as with Docker, we need to select image to preload correct versions of postgres. Here it is `postgres:9.6.2-alpine`. And set environment variables as from `config/database.yml`
+```
+    build:
+        working_directory: ~/your-app-name
+        docker:
+            - image: circleci/ruby:2.5.1-node-browsers
+            environment:
+                RAILS_ENV: test
+
+            - image: circleci/postgres:9.6.2-alpine
+            environment:
+                POSTGRES_USER: postgres
+                POSTGRES_DB: test_database_name
+```
+
+## Process running steps
+Starting from this. All commands go in the `steps` key. Which begin with:
+```
+    steps:
+      - checkout
+```
+
+### Prepare environment
```

Comment:
### Prepare your Environment

We almost always use Figaro for storing and loading environment variables, so firstly we must create a copy of the `application.yml` file: